### PR TITLE
GenesisConfig for Next_Length_multi & Target_Block_Size

### LINKED ISSUE #178: fees: use GenesisConfig in pallet length-fee-adjustment
Currently, the pallet length-fee-adjustment uses default values when empty for the storage items NextLengthMultiplier and TargetBlockSize. This is done to avoid dealing with some downsides of using GenesisConfig to define the initial values of those storage items.

The cleanest solution, however, would be to use a custom GenesisConfig in the pallet. This would make the pallet more usable for other projects as well.

Switching to this approach would likely require addressing all the necessary steps for handling a storage migration.

### PR DESCRIPTION
**Pull request by MrishoLukamba**
This PR Fixes #178.

It intrudces optional set up for those 2 storage values,

if not provided the default ones will be in use

### CODE REVIEW

**Review by rphmeier — COMMENTED** [review]
Thanks. Can you add some docs to the new `pub` items?

**Comment by MrishoLukamba** [comment]
> Thanks. Can you add some docs to the new `pub` items?

Yes

File: sugondat/chain/pallets/length-fee-adjustment/src/lib.rs [hunk 1]
```diff
@@ -139,6 +140,30 @@ pub mod pallet {
     pub type TargetBlockSize<T: Config> =
         StorageValue<_, Perquintill, ValueQuery, TargetBlockSizeDefault>;
 
+    /// Genesis config for setting up `NextLengthMultiplier` and `TargetBlockSize` storage.
+    /// Set None if the default values `NextLengthMultiplierDefault` & `TargetBlockSizeDefault` are to be used.
+    #[pallet::genesis_config]
+    #[derive(frame_support::DefaultNoBound)]
+    pub struct GenesisConfig<T: Config> {
+        /// Genesis set up for next length multiplier storage item
+        pub next_length_multiplier: Option<FixedU128>,
```
Comment (gabriele-0201) [inline sugondat/chain/pallets/length-fee-adjustment/src/lib.rs:149]: `Multiplier` is an alias for `FixedU128`, it is used by `pallet_transaction_payment` and was also used in the file. Therefore, it should be used here as well

File: sugondat/chain/runtimes/test/src/lib.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ use sp_runtime::{
     transaction_validity::{TransactionSource, TransactionValidity},
     ApplyExtrinsicResult, FixedPointNumber, Perquintill,
 };
+pub use pallet_transaction_payment::Multiplier;
```
Comment (gabriele-0201) [inline sugondat/chain/runtimes/test/src/lib.rs:24]: Why did you remove the `pub use` from the other line and add it in a new line?

**Review by gabriele-0201 — DISMISSED** [review]

File: sugondat/chain/runtimes/test/src/lib.rs (hunk 2 above)
Comment (MrishoLukamba) [reply sugondat/chain/runtimes/test/src/lib.rs:24]: I wanted to re export only whats needed which is the Multiplier

Comment (MrishoLukamba) [reply sugondat/chain/runtimes/test/src/lib.rs]: Yes but again we will define a new type alias multiplier for consistency

Comment (rphmeier) [reply sugondat/chain/runtimes/test/src/lib.rs]: We should re-export from primitives and use it here & in node rather than re-exporting from the runtimes.

File: sugondat/chain/pallets/length-fee-adjustment/src/lib.rs [hunk 3]
```diff
@@ -139,6 +140,30 @@ pub mod pallet {
     pub type TargetBlockSize<T: Config> =
         StorageValue<_, Perquintill, ValueQuery, TargetBlockSizeDefault>;
 
+    /// Genesis config for setting up `NextLengthMultiplier` and `TargetBlockSize` storage values.
+    /// Set None if the default values `NextLengthMultiplierDefault` & `TargetBlockSizeDefault` are to be used.
```
Comment (rphmeier) [inline sugondat/chain/pallets/length-fee-adjustment/src/lib.rs:144]: Having the `_Default` members was a workaround for _not_ having a genesis config and migration. They could be removed in the future. The fields of `GenesisConfig` should not be `Option`s.

Comment (MrishoLukamba) [reply sugondat/chain/pallets/length-fee-adjustment/src/lib.rs]: Done

**Review by rphmeier — APPROVED** [review]
Thanks for the PR!
